experiments/download_ssl4eo-s12.py

```diff
@@ -0,0 +1,718 @@
+""" Sample and download Sentinel-1/2 tiles with Google Earth Engine
+
+#### run the script:
+# resample new ids with grid overlap search
+!python ssl4eo_s12_downloader.py \
+    --save_path ./data \
+    --num_workers 8 \
+    --cloud_pct 20 \
+    --log_freq 100 \
+    --indices_range 0 250000
+
+### resume from interruption
+!python ssl4eo_s12_downloader.py \
+    --save_path ./data \
+    --num_workers 8 \
+    --cloud_pct 20 \
+    --log_freq 100 \
+    --resume ./data/checked_locations.csv \
+    --indices_range 0 250000
+
+"""
+
+import argparse
+import csv
+import json
+import math
+import os
+import time
+import warnings
+from collections import OrderedDict
+from datetime import date, timedelta
+from multiprocessing.dummy import Lock, Pool
+
+import ee
+import numpy as np
+import rasterio
+import shapefile
+import urllib3
+from rasterio.transform import Affine
+from shapely.geometry import Point, shape
+
+# from skimage.exposure import rescale_intensity
+from torchvision.datasets.utils import download_and_extract_archive
+from tqdm import tqdm
+
+warnings.simplefilter("ignore", UserWarning)
+
+ALL_BANDS_L2A = [
+    "B1",
+    "B2",
+    "B3",
+    "B4",
+    "B5",
+    "B6",
+    "B7",
+    "B8",
+    "B8A",
+    "B9",
+    "B11",
+    "B12",
+]
+ALL_BANDS_L1C = [
+    "B1",
+    "B2",
+    "B3",
+    "B4",
+    "B5",
+    "B6",
+    "B7",
+    "B8",
+    "B8A",
+    "B9",
+    "B10",
+    "B11",
+    "B12",
+]
+RGB_BANDS = ["B4", "B3", "B2"]
+ALL_BANDS_GRD = ["VV", "VH"]
+
+
+""" samplers to get locations of interest points"""
+
+
+class GeoSampler:
```

Comment:
We might want to rename this to avoid confusion with `torchgeo.samplers.GeoSampler`